spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/mail/JndiMailAutoConfiguration.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2012-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.mail;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jndi.JndiLocatorDelegate;
+import org.springframework.mail.MailMessage;
+import org.springframework.util.StringUtils;
+
+import javax.mail.Session;
+import javax.naming.NamingException;
+
+/**
+ * @author Eddú Meléndez
+ */
+@Configuration
+@AutoConfigureBefore(MailSenderAutoConfiguration.class)
+@ConditionalOnClass(MailMessage.class)
+@ConditionalOnProperty(prefix = "spring.mail", name = "jndi-name")
+@EnableConfigurationProperties(MailProperties.class)
+public class JndiMailAutoConfiguration {
```

Comment:
I don't think it's worth a separate auto-config class. After all you are configuring something that is related to mail in a different flavour. How about moving that class as a nested static class of `MailSenderAutoConfiguration`. Please look at `JmsAnnotationDrivenConfiguration` for an example that does the same thing with `DestinationResolver`.